packages/evm/test/precompiles/05-modexp.spec.ts

```diff
@@ -10,6 +19,26 @@ import type { PrefixedHexString } from '@ethereumjs/util'
 import type { EVM } from '../../src/index.ts'
 import type { PrecompileFunc } from '../../src/precompiles/types.ts'
 
+/**
+ * Compute base^exp mod modulus for *arbitrarily* large BigInts
+ */
+function modPow(base: bigint, exp: bigint, modulus: bigint): bigint {
```

Comment:
Can't this be done directly with bigints?

Reply:
When exponentiation is done natively with bigints using the `**` operator, the value that's being raised to a power is literally formed in memory (base×base×⋯×base), which for very large numbers is astronomically huge and will quickly blow up with “Maximum BigInt size exceeded”. This helper function uses the binary exponentiation algorithm, similar to the actual precompile implementation, in order to get around that.